tests/basic/pthread_detach/README.md

```diff
@@ -0,0 +1,181 @@
+From the LKL paper -
+> To regain control of scheduling, the generic LKL architecture layer associates
+> an environment-provided semaphore with each LKL thread.
+>
+> Immediately after creation, and before running any Linux code each LKL thread
+> acquires its corresponding semaphore, and gets blocked as the semaphore's
+> initial value is 0.
+>
+> When the Linux scheduler selects a new thread to run, it releases the semaphore
+> of the new thread and immediately acquires its own semaphore.
+>
+> The new thread will start running and the old one stops.
+>
+> This token passing mechanism ensures that at any given time there is only one
+> thread running and the scheduling order is dictated by the Linux scheduler.
+
+O. Purdila, L. A. Grijincu and N. Tapus, "LKL: The Linux kernel library," 9th RoEduNet IEEE International Conference, Sibiu, 2010, pp. 328-333.
+
+The environment-provided semaphore is the `sched_sem` field in thread_info.
+
+## thread_sched_jb()
+`thread_sched_jb` is LKL's way of doing co-operative multitasking, by yielding to the scheduler. Its invoked from - 
+
+1. `lkl_cpu_put`: If the Linux scheduler run queue has >= 1 tasks.
+2. `switch_to_host_task(task)`: If `current` process context != task.
+3. `lkl_syscall`: If syscall is NR_REBOOT, yields to scheduler after running the syscall.
+
+
+## switch_to_host_task(task)
+`switch_to_host_task` function ensures that task being passed as input parameter is the `current` process context. If this is already the case the function returns early. Else, this causes transfer of control to the LKL scheduler so that it can schedule `task`. 
+A flag(TIF_SCHED_JB) is added to the task's flags before yielding to the LKL scheduler. This helps identify this task from the other tasks in the architecture dependent switching routine `__switch_to`. Once the thread is in `__switch_to`, control is switched back to `switch_to_host_task` via `threads_sched_jb`. After that the thread sleeps on its scheduler semaphore and waits for the LKL scheduler to wake it.
+
+Call graph for switch_to_host_task to __switch_to interactions:
+```
+switch_to_host_task(task: task_struct)
+    - task.ti.tid = lthread_self()
+    - wake_up_process(task)
+    |
+    V
+    - thread_sched_jb()
+        - current.ti.flags = TIF_SCHED_JB
+        - current.state = TASK_UNINTERRUPTIBLE
+    (long jump to scheduler)
+    |
+    V
+    schedule
+        |
+        V
+        - __schedule(preempt: bool)
+            - prev = rq->current
+            - next = pick_next_task()
+            |
+            V
+            - context_switch(rq, prev, next)
+                |
+                V
+                - __switch_to(prev, next)
+                    - set abs_prev
+                    - change cpu ownership to next
+                    - Wake lthread backing _next
+                    - As prev has TIF_SCHED_JB flag set, long jump back.
+    |
+    V
+    (back in threads_sched_jb())
```

Comment:
This might be easier to understand if one indent corresponded to one call depth.